llarp/crypto/encrypted_frame.hpp

```diff
@@ -78,12 +77,14 @@ namespace llarp
     const SecretKey& seckey;
     EncryptedFrame target;
 
+    using WorkFunc_t = std::function<void(void)>;
+    using WorkerFunction_t = std::function<void(WorkFunc_t)>;
+
     void
-    AsyncDecrypt(
-        const std::shared_ptr<thread::ThreadPool>& worker, const EncryptedFrame& frame, User_ptr u)
+    AsyncDecrypt(const EncryptedFrame& frame, User_ptr u, WorkerFunction_t worker)
     {
       target = frame;
-      worker->addJob(std::bind(&AsyncFrameDecrypter<User>::Decrypt, this, std::move(u)));
+      worker(std::bind(&AsyncFrameDecrypter<User>::Decrypt, this, std::move(u)));
```

Comment:
Since this is only called once, we can avoid the extra shared_ptr ref count increment in Decrypt via a mutable lambda:
```C++
worker([this, u=std::move(u)] () mutable { Decrypt(std::move(u)); });
```